src/test/java/com/puppycrawl/tools/checkstyle/internal/ArchUnitSuperClassTest.java

```diff
@@ -0,0 +1,165 @@
+import com.tngtech.archunit.core.importer.ImportOption;
+import com.tngtech.archunit.lang.ArchCondition;
+import com.tngtech.archunit.lang.ArchRule;
+import com.tngtech.archunit.lang.ConditionEvents;
+import com.tngtech.archunit.lang.SimpleConditionEvent;
+
+public class ArchUnitSuperClassTest {
+
+    /**
+     * Classes not abiding to {@code checksShouldExtendCorrectAbstractClass} rule.
+     */
+    private static final Set<String> SUPPRESSED_CLASSES = Set.of(
+        "com.puppycrawl.tools.checkstyle.checks.coding.SuperCloneCheck",
+        "com.puppycrawl.tools.checkstyle.checks.coding.SuperFinalizeCheck",
+        "com.puppycrawl.tools.checkstyle.checks.header.HeaderCheck",
+        "com.puppycrawl.tools.checkstyle.checks.header.RegexpHeaderCheck",
+        "com.puppycrawl.tools.checkstyle.checks.metrics.ClassDataAbstractionCouplingCheck",
+        "com.puppycrawl.tools.checkstyle.checks.metrics.ClassFanOutComplexityCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.AbstractAccessControlNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.CatchParameterNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.ClassTypeParameterNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.ConstantNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.IllegalIdentifierNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.InterfaceTypeParameterNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.LambdaParameterNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.LocalFinalVariableNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.LocalVariableNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.MemberNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.MethodNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.MethodTypeParameterNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.ParameterNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.PatternVariableNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.RecordComponentNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.RecordTypeParameterNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.StaticVariableNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.naming.TypeNameCheck",
+        "com.puppycrawl.tools.checkstyle.checks.whitespace.ParenPadCheck",
+        "com.puppycrawl.tools.checkstyle.checks.whitespace.TypecastParenPadCheck"
+    );
+
+    /**
+     * ArchCondition checking that a class is a subclass of a given class.
+     *
+     * @param superclass the superclass
+     * @return ArchCondition checking that a class is a subclass of a given class
+     */
+    private static ArchCondition<JavaClass> beSubclassOf(Class<?> superclass) {
+        return new SuperclassArchCondition(superclass);
```

Comment:
beSubclassOf -> hasSupperClass

or may be you have better name.

Reply:
Done.